.github/workflows/documentation-pr.yml

```diff
@@ -45,15 +45,29 @@ jobs:
             const enumeration = JSON.parse(JSON.parse(ENUMERATION));
                                     
             if (!enumeration.includes('plasma-core')) {
-              enumeration.push('plasma-core')
+              enumeration.push('plasma-core');
             }
             
             if (!enumeration.includes('plasma-hope')) {
-              enumeration.push('plasma-hope')
+              enumeration.push('plasma-hope');
             }
             
             if (!enumeration.includes('plasma-icons')) {
-              enumeration.push('plasma-icons')
+              enumeration.push('plasma-icons');
+            }
+            
+            if (!enumeration.includes('plasma-docs-ui')) {
+              enumeration.push('plasma-docs-ui');
+            }
+            
+            if (!enumeration.includes('plasma-website')) {
+              enumeration.push('plasma-website');
+            }
+            
+            if (enumeration.includes('plasma-temple-docs') || enumeration.includes('plasma-ui-docs')) {
+              enumeration.push('plasma-web');
+            
+              enumeration.push('plasma-b2c');
             }
```

Comment:
У меня ощущение что мы движемся куда то не туда, как будто у нас есть граф зависимостей который описан через package.json & lerna.json а есть ещё один параллельный тут :/

Reply:
@Yeti-or Мне тоже не нравятся все эти блоки с `if`

В данном моменте почему так получилось

нам всегда нужно что бы собирался пакет `plasma-website`

Для того чтобы работали `plasma-temple-docs` и  `plasma-ui-docs` нужны пакеты: 

1. @salutejs/plasma-docs-ui 
2. @salutejs/plasma-web
3.@salutejs/plasma-b2c <-- вот с этим непонятно

Reply:
Если смотреть в сторону запуска с флагом `--include-dependencies` 

Это результат для ветки - "neretinaa/fix-scrollable-behavior-sheet-wip" 

```bash
{"name":"@salutejs/plasma-temple","version":"1.167.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-temple"}
{"name":"@salutejs/plasma-ui","version":"1.199.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-ui"}
{"name":"@salutejs/plasma-temple-docs","version":"0.158.0","private":true,"location":"/Users/workspace/plasma/website/plasma-temple-docs"}
{"name":"@salutejs/plasma-ui-docs","version":"0.207.0","private":true,"location":"/Users/workspace/plasma/website/plasma-ui-docs"}
{"name":"@salutejs/plasma-cy-utils","version":"0.64.0","private":false,"location":"/Users/workspace/plasma/utils/plasma-cy-utils"}
{"name":"@salutejs/plasma-icons","version":"1.152.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-icons"}
{"name":"@salutejs/plasma-tokens","version":"1.57.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-tokens"}
{"name":"@salutejs/plasma-sb-utils","version":"0.121.0","private":false,"location":"/Users/workspace/plasma/utils/plasma-sb-utils"}
{"name":"@salutejs/plasma-core","version":"1.123.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-core"}
{"name":"@salutejs/plasma-typo","version":"0.35.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-typo"}
{"name":"@salutejs/plasma-docs-ui","version":"0.133.0","private":true,"location":"/Users/workspace/plasma/utils/plasma-docs-ui"}
{"name":"@salutejs/plasma-tokens-web","version":"1.38.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-tokens-web"}
{"name":"@salutejs/plasma-web","version":"1.218.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-web"}
{"name":"@salutejs/plasma-tokens-b2b","version":"1.23.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-tokens-b2b"}
{"name":"@salutejs/plasma-tokens-b2c","version":"0.32.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-tokens-b2c"}
{"name":"@salutejs/plasma-colors","version":"0.8.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-colors"}
{"name":"@salutejs/plasma-tokens-utils","version":"0.28.0","private":false,"location":"/Users/workspace/plasma/utils/plasma-tokens-utils"}
{"name":"@salutejs/plasma-tokens-core","version":"0.3.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-tokens-core"}
{"name":"@salutejs/plasma-hope","version":"1.218.0","private":false,"location":"/Users/workspace/plasma/packages/plasma-hope"}
```

Флаг `--ignore` в сочетание с `--include-dependencies` не особо работает. 

> Note: This will override the --scope and --ignore flags.

> i.e. A package matched by the --ignore flag will still be bootstrapped if it is depended on by another package that is being bootstrapped.

[docs](https://www.npmjs.com/package/@lerna/filter-options)

Reply:
давай еще раз мы хотим запускать plasma-website для него нужно куча зависимостей, но их все мы ставить не хотим? хотим выборочно ?

Reply:
@Yeti-or 

> мы хотим запускать plasma-website для него нужно куча зависимостей, но их все мы ставить не хотим? хотим выборочно ?

Да, именно так. 

Для установки `plasma-website` требуется ряд других пакетов. 

И эти зависимости нужно ставить **явно**, если их нет в списки измененных.

------ 

Вчера общались с @neretin-trike на эту тему, и вполне возможно, что **декларативное** описание графа зависимостей и есть наш вариант.  

Что-то типо 

```json
{
"plasma-website": ["plasma-web", "plasma-b2c", "etc"],
"plasma-ui-docs": ["plasma-web", "plasma-docs-ui", "etc"]
}
```

Reply:
а мы можем парсить package.json в CI? Просто мне кажется, что эти зависимости можно было бы добавить на уровне пакетов дополнительным полем типа:

```JSON
{
    "name": "@salutejs/plasma-ui-docs",
    "version": "0.207.0",
    "description": "Plasma Documentation.",
    .....
    "continuousIntegrationDependencies": {
        "@salutejs/plasma-web": "9.9.9",
        "@salutejs/plasma-b2c": "9.9.9",
        ....
    }
}

```

Можно не прям сейчас, но в будущем 🤔